src/transformers/models/grounding_dino/convert_grounding_dino_to_hf.py

```diff
@@ -0,0 +1,426 @@
+        window_size = 12
+        embed_dim = 128
+        depths = (2, 2, 18, 2)
+        num_heads = (4, 8, 16, 32)
+        image_size = 384
+    else:
+        raise ValueError("Model not supported, only supports base and large variants")
+
+    config.backbone_config.window_size = window_size
+    config.backbone_config.image_size = image_size
+    config.backbone_config.embed_dim = embed_dim
+    config.backbone_config.depths = depths
+    config.backbone_config.num_heads = num_heads
+    config.backbone_config.out_indices = [2, 3, 4]
+
+    return config
+
+
+def create_rename_keys(state_dict, config):
+    rename_keys = []
+    # fmt: off
+    #TODO names might change after modifing GroundingDINOModel class
+    ########################################## VISION BACKBONE - START
+    # patch embedding layer
+    rename_keys.append(("backbone.0.patch_embed.proj.weight",
+                        "model.backbone.conv_encoder.model.embeddings.patch_embeddings.projection.weight"))
+    rename_keys.append(("backbone.0.patch_embed.proj.bias",
+                        "model.backbone.conv_encoder.model.embeddings.patch_embeddings.projection.bias"))
+    rename_keys.append(("backbone.0.patch_embed.norm.weight",
+                        "model.backbone.conv_encoder.model.embeddings.norm.weight"))
+    rename_keys.append(("backbone.0.patch_embed.norm.bias",
+                        "model.backbone.conv_encoder.model.embeddings.norm.bias"))
+
+    for layer, depth in enumerate(config.backbone_config.depths):
+        for block in range(depth):
+            # layernorms
+            rename_keys.append((f"backbone.0.layers.{layer}.blocks.{block}.norm1.weight",
+                                f"model.backbone.conv_encoder.model.encoder.layers.{layer}.blocks.{block}.layernorm_before.weight"))
+            rename_keys.append((f"backbone.0.layers.{layer}.blocks.{block}.norm1.bias",
+                                f"model.backbone.conv_encoder.model.encoder.layers.{layer}.blocks.{block}.layernorm_before.bias"))
+
+            rename_keys.append((f"backbone.0.layers.{layer}.blocks.{block}.norm2.weight",
+                                f"model.backbone.conv_encoder.model.encoder.layers.{layer}.blocks.{block}.layernorm_after.weight"))
+            rename_keys.append((f"backbone.0.layers.{layer}.blocks.{block}.norm2.bias",
+                                f"model.backbone.conv_encoder.model.encoder.layers.{layer}.blocks.{block}.layernorm_after.bias"))
+            # attention
+            rename_keys.append((f"backbone.0.layers.{layer}.blocks.{block}.attn.relative_position_bias_table",
+                                f"model.backbone.conv_encoder.model.encoder.layers.{layer}.blocks.{block}.attention.self.relative_position_bias_table"))
+            # rename_keys.append((f"backbone.0.layers.{layer}.blocks.{block}.attn.relative_position_index",
+            #                     f"encoder.layers.{layer}.blocks.{block}.attention.relative_position_index"))
```

Comment:
if not used, it would be better to remove

Reply:
FIxed here https://github.com/huggingface/transformers/pull/26087/commits/e14d6aea74f9857cfb583c2a05c7eeb7785b3e0d